broker.go

```
@@ -237,13 +237,18 @@ func Dial(ctx context.Context, nodeAddresses []string, conf BrokerConf) (*Broker
 			conf.Logger.Debug("cannot fetch metadata from any connection",
 				"retry", i,
 				"sleep", conf.DialRetryWait)
-			time.Sleep(conf.DialRetryWait)
+			select {
+			case <-time.After(conf.DialRetryWait):
+				// continue
+			case <-ctx.Done():
+				return nil, ctx.Err()
+			}
 		}
 
-		err := broker.refreshMetadata(ctx, proto.MetadataV0)
-
-		if err == nil {
+		if err := broker.refreshMetadata(ctx, proto.MetadataV0); err == nil {
 			return broker, nil
+		} else if err := ctx.Err(); err != nil {
```

This seems to swallow the error from line 248, is this intended?

---

Yes. It means: If there is an error and the context is canceled, ignore the error and return the cancelation

---

OK.